abci/src/lib.rs

```diff
@@ -0,0 +1,21 @@
+//! ABCI framework for building Tendermint applications in Rust.
+
+mod application;
+#[cfg(feature = "client")]
+mod client;
+mod codec;
+mod result;
+mod server;
+
+// Common exports
+pub use application::Application;
+#[cfg(feature = "client")]
+pub use client::{Client, ClientBuilder};
+pub use result::{Error, Result};
+pub use server::{Server, ServerBuilder};
+
+// Example applications
```

Comment:
Do we need to expose these apps?

Reply:
Looking at this again now, the integration tests need these apps to be exposed, and they might actually be useful as components of test suites (e.g. a full node will need to import an application like the kvstore for in-process testing).